server/handler/base.go

```diff
@@ -33,14 +33,28 @@ type Base struct {
 	PushRestrictionUserToken string
 }
 
-func (b *Base) ProcessPullRequest(ctx context.Context, pullCtx pull.Context, client *github.Client, pr *github.PullRequest) error {
+func (b *Base) FetchConfig(ctx context.Context, client *github.Client, pr *github.PullRequest) (*bulldozer.Config, error) {
 	logger := zerolog.Ctx(ctx)
 
 	bulldozerConfig, err := b.ConfigForPR(ctx, client, pr)
 	if err != nil {
-		return errors.Wrap(err, "failed to fetch configuration")
+		return nil, errors.Wrap(err, "failed to fetch configuration")
 	}
 
+	switch {
+	case bulldozerConfig.Missing():
+		return nil, errors.Errorf("No configuration found for %s", bulldozerConfig)
```

Comment:
One consequence of this conversion is that these conditions that were previously only logged are now returned and propagated as errors. I think I'd like to keep the old behavior by logging the same messages here and then returning `nil, nil`.

This means that either callers will have to check for `nil` or `UpdatePR` and `ProcessPR` should return immediately if passed a `nil` configuration; I'd probably have the functions return immediately.